vtex/node-vtex-api
# Feature/service configuration context

#### What is the purpose of this pull request?
Add the settings middleware. With this modifications, services will receive their configurations from other apps in the request context.

#### What problem is this solving?
Better interface for getting configurations.

#### How should this be manually tested?
Create a service with `workspace` or `userAndWorkspace` settings type. After that, create an app with configurations to be exported to this service. Link them all and see if the request context comes with the configurations.

In more detail, in your `service` you must have:

1. A route with settings type setted as `workspace` or `userAndWorkspace`. For that, add something like this to your `node/service.json`:
```
"status": {
      "path": "/_v/status/:code",
      "public": true,
      "settingsType": "workspace"
    }
```

2. In the `manifest.json`, don't forget to add the `configuration` builder:
```
"builders": {
    "configuration": "0.x",
    ...
  },
...
```

3. Add the file `configuration/schema.json` with a schema for the configuration, for instance:
```
{
  "type": "object",
  "properties": {
    "id": { "type": "number" },
    "name": { "type": "string" }
  }
}
```

In your `app` that exports the configuration, you must have:
1. In the `manifest.json`, a declaration of the service as a builder:
```
"builders": {
    "vtex.<serviceName>": "0.x",
    ...
  },
```

2. The file `vtex.<serviceName>/configuration.json` must have a configuration that conforms to schema in the service:
```
{
  "name": "little foot",
  "id": 19
}
```

Link first the service, and then the app. Make the request and check in `ctx.vtex.settings` if there are the exported settings

#### Types of changes
